src/Featurizers/ShortGrainDropperFeaturizer.h

```diff
@@ -192,9 +192,9 @@ ShortGrainDropperEstimator<MaxNumTrainingItemsV>::ShortGrainDropperEstimator(
             //cv may not be 0, not sure currently
             if (std::distance(std::get<0>(lags), std::get<1>(lags)) == 0)
                 throw std::invalid_argument("lags");
-            if (!cv.has_value())
-                return (maxHorizon + std::max(windowSize, *std::max_element(std::get<0>(lags), std::get<1>(lags))) + 1);
-            return (2*maxHorizon + static_cast<std::uint8_t>(*cv) + std::max(windowSize, *std::max_element(std::get<0>(lags), std::get<1>(lags))) + 1);
+            if (!crossValidation.has_value())
+                return (horizon + std::max(maxWindowSize, *std::max_element(std::get<0>(lags), std::get<1>(lags))) + 1);
+            return (2*horizon + *crossValidation + std::max(maxWindowSize, *std::max_element(std::get<0>(lags), std::get<1>(lags))) + 1);
```

Comment:
I don't understand what this code is doing. Jin and I have spent a lot of time talking about a similar topic with regards to the LagLeadOperator featurizer. Can you sync with him and try to make this code simpler.

Reply:
sure